packages/loader/container-loader/src/container.ts

```diff
@@ -1128,6 +1138,7 @@ export class Container
 			}
 		} finally {
 			this._lifecycleState = "closed";
+			this._criticalError = error;
```

Comment:
i wonder if we should push this up higher, so it is observable within events, like `closed`

Reply:
Yeah that makes sense.  I'll push it up to the state transition to "closing"/"disposing".